quartz/components/scripts/popover.inline.ts

```diff
@@ -82,8 +82,13 @@ async function mouseEnterHandler(
       const contents = await response.text()
       const html = p.parseFromString(contents, "text/html")
       normalizeRelativeURLs(html, targetUrl)
-      // strip all IDs from elements to prevent duplicates
-      html.querySelectorAll("[id]").forEach((el) => el.removeAttribute("id"))
+      // prepend all IDs inside popovers to prevent duplicates
+      html.querySelectorAll("[id]").forEach((el) => {
+        const targetID = `popover-${el.id}`
+        el.id = targetID
+      })
+      const targetLink = hash.startsWith("#popover") ? hash : `#popover-${hash.slice(1)}`
```

Comment:
not all links are popovers right 👀 this seems to prepend popover to all anchors

Reply:
It _does_ prepend popover to all anchors, but only anchors inside popover overlays. This is to ensure no duplicate `id` exist (e.g. linking to a heading/block reference on the current page.)

Reply:
we should avoid reassigning hash then and just change L102 to do ``popoverInner.getElementById(`popover-${hash}`)``

Reply:
`HTMLDivElement` does not have a method `getElementById()`. I have made a change to accomplice the same effect.